source/operatingsystems/linux/mysql/mysql_ssh_tunneling.md

```diff
@@ -0,0 +1,89 @@
+
+In this guide we'll discuss how you can create a SSH tunnel to connect to MySQL remotely without the need to open MySQL to the public. 
+
+## What is SSH Tunnelling
+SSH tunnelling allows a remote user to redirect traffic from one network port to another. It creates a tunnel between a client computer and a server over the encrypted channel SSH, allowing remote users to pass traffic to the server as though the traffic originated locally to the server. This becomes extremely useful when connecting to services such as `MySQL`, as it's best practice to keep the MySQL port closed and MySQL bound to localhost only. 
+
+## SSH Tunnelling with MySQL Workbench
+
+### Creating a new connection
+- Start by opening MySQL Workbench
+-  Next to MySQL Connection, click the + button. 
+
+Below are the Parameters you will need to fill in:
+
+| Parameters | Description |
+|-|-|
+| Connection Name | A name for the connection that uniquely identifies the connection you're making.  |
+| SSH Hostname | The IP address of the server you're connecting to, followed by the port. With UKFast installations, the default SSH port is 2020. For example, `1.2.3.4:2020` |
+| SSH Username | The username you use to SSH onto the server. |
+| SSH Password/SSH Key File | This is the SSH password or the SSH key for your server. |
+| MySQL Hostname | As we're connecting via an SSH Tunnel, this will likely be 127.0.0.1 |
+| MySQL Server Port | The default for this is 3306 |
+| Username | The MySQL user that you're connecting as. As we're connecting locally, it will need to be a user which has the host set to localhost.  |
+| Password | The MySQL password for the MySQL user.  |
+
+```eval_rst
+.. seealso::
+   Note:
+     When entering passwords, MySQL Workbench offers users the ability to store passwords in a Vault, which is encrypted on your machine. See [MySQL - The Password Storage Vault](https://dev.mysql.com/doc/workbench/en/wb-mysql-connections-vault.html) for more information.
+```
+
+### Testing the connection
+- Once you've entered all the above parameters, click `Test Connection`.
+- You may receive a prompt about the fingerprint not being recognised. If this is the first time connecting to the server and you're confident that the details you've entered are correct, click **Yes**. 
+
+This should return **Successfully made the MySQL connection**. Click **Ok**, and then **Ok** again to save the configuration. If this is unsucessful due to incorrect MySQL credentials, check that the user you're connecting with is set with the host `localhost`.
+
+## SSH Tunnelling with Other Clients (Windows)
+While most MySQL clients support SSH tunnelling, some clients do not. Using PuTTY, we can create an SSH Tunnel systemwide, so that you can connect to MySQL on localhost:3306, which will route the traffic to the server. To do this, follow the following steps: 
```

Comment:
follow the following steps -> use the following steps / follow the below steps ?